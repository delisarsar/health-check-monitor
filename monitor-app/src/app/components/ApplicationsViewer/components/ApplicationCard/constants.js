import React from "react";
import { FormattedMessage } from "react-intl";

export const EDIT_APPLICATION_LABEL = (
  <FormattedMessage id="overview.edit" defaultMessage={"Edit"} />
);

export const ENDPOINT_APPLICATION_LABEL = (
  <FormattedMessage id="overview.endpoint" defaultMessage={"Endpoint"} />
);
