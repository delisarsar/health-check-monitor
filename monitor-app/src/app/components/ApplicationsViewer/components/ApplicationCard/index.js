import React, { memo } from "react";
import PropTypes from "prop-types";
import Card from "@material-ui/core/Card";
import { Typography } from "@material-ui/core";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import Button from "@material-ui/core/Button";
import useStyles from "./styles";
import {
  EDIT_APPLICATION_LABEL,
  ENDPOINT_APPLICATION_LABEL,
} from "./constants";

const ApplicationCard = ({ id, appName, healthCheckEndpoint, onChange }) => {
  const classes = useStyles();
  return (
    <Card
      data-testid={`application-card-${id}`}
      key={id}
      elevation={2}
      className={classes.root}
    >
      <CardContent className={classes.cardContent}>
        <Typography data-testid="card-app-name">{appName}</Typography>
        <Typography data-testid="card-endpoint">
          {ENDPOINT_APPLICATION_LABEL}: {healthCheckEndpoint}
        </Typography>
      </CardContent>
      <CardActions className={classes.cardActions}>
        <Button
          data-testid="edit-button"
          onClick={() => onChange(id)}
          className={classes.hintActionButton}
        >
          {EDIT_APPLICATION_LABEL}
        </Button>
      </CardActions>
    </Card>
  );
};

export default memo(ApplicationCard);

ApplicationCard.propTypes = {
  id: PropTypes.number.isRequired,
  appName: PropTypes.string.isRequired,
  healthCheckEndpoint: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};
