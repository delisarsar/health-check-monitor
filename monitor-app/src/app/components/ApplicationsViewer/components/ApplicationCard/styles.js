import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  root: {
    width: 700,
    height: 150,
  },
  content: {
    color: theme.palette.primary.main,
  },
  cardActions: {
    paddingLeft: theme.spacing(37.5),
    paddingTop: 0,
    marginBottom: theme.spacing(3),
  },
  cardContent: {
    paddingTop: theme.spacing(2),
    paddingBottom: 0,
  },
}));

export default useStyles;
