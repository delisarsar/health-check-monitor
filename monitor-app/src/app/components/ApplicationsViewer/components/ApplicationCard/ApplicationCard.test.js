import React from "react";
import userEvent from "@testing-library/user-event";
import { render } from "@testing-library/react";
import { IntlProvider } from "react-intl";
import ApplicationCard from ".";
import messages from "../../../../../translations/en.json";

test("renders correctly", () => {
  const { container } = render(
    <IntlProvider locale="en-US" messages={messages}>
      <ApplicationCard
        id={1}
        appName="name"
        healthCheckEndpoint="endpoint"
        onChange={() => jest.fn()}
      />
    </IntlProvider>
  );
  expect(container).toMatchSnapshot();
});

test("navigate call the edit app function on click ", () => {
  const mockCallback = jest.fn();
  const { getByTestId } = render(
    <IntlProvider locale="en-US" messages={messages}>
      <ApplicationCard
        id={1}
        appName="name"
        healthCheckEndpoint="endpoint"
        onChange={mockCallback}
      />
    </IntlProvider>
  );
  const editButton = getByTestId("edit-button");
  userEvent.click(editButton);
  expect(mockCallback).toHaveBeenCalled();
});
