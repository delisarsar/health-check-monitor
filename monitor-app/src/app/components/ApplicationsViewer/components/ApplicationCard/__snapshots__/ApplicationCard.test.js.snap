// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation2 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root makeStyles-cardContent-4"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
        data-testid="card-app-name"
      >
        name
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1"
        data-testid="card-endpoint"
      >
        Endpoint
        : 
        endpoint
      </p>
    </div>
    <div
      class="MuiCardActions-root makeStyles-cardActions-3 MuiCardActions-spacing"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        data-testid="edit-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Edit
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
